docs/development/pigs/pig-026.rst

```diff
@@ -0,0 +1,328 @@
+the estimated IRFs systematics provided by the corresponding experiment, and yield more realistic and trustworthy results. 
+The proposed formalism also includes the application of nuisance parameters and regularization.
+
+Use cases
+=========
+
+In the past priors were a regularly requested feature or the solution to a problem.
+See the following issues and PRs of Gammapy:
+
+- Add sigma v estimator functionality for dark matter use case `#2075`_
+- Background systematics as a nuisance parameter `#3955`_
+- Support unfolding methods for spectral flux points `#4122`_
+- Add PiecewiseNormSpatialModel `#4208`_
+
+Implementation
+==============
+
+The following implementation draft is compatible with the current API,
+where the models are set via Dataset.model and the Fit is run via Fit.run(datasets).
+
+Prior (abstract) base class:
+--------------
+
+.. code:: python 
+
+  class Prior():
+  
+     def __call__(self, base):
+     
+        if isinstance(base, Parameter):
+            return self.evaluate(base.value)      
+        else: # model-like instance
+            return self.evaluate(base.parameters.value)  
+         
+
+Future additional properties of the ``Prior`` classes: 
+
+- Fittable parameters
+- Write/read from/to a yaml file, ideally also when the corresponding model is written/read (see serialisation example below)
+- Prior registry system
+- Prior subclass use cases (i.e. for correlated nuisance parameters with correlation matrix as a parameter)
+
+Exemplary additional prior subclasses:
+--------------
+.. code:: python
+
+  class GaussianPrior(Prior):
+  
+      """Gaussian Prior with mu and sigma set on a single parameter
+      """
+      
+      def __init__(self, mu=0, sigma=1):
+          self.mu = mu
+          self.sigma = sigma
+
+      def evaluate(self, value):
+          return ((value - self.mu) / self.sigma) ** 2
+
+
+.. code:: python
+
+    class UniformPrior(Prior):
+    
+        """Uniform Prior set on a single parameter or model.
+        """
+        
+        def __init__(self, uni=0):
+            self.uni = uni
+
+        def evaluate(self, value):
+            return self.uni
+
+.. code:: python 
+  
+    class CovarianceGaussianPrior(Prior):
+        """Gaussian Priors on mulitple parameters with covariance matrix. Set on a model. 
+        
+        Parameters
+        ----------
+        cov : `~numpy.ndarray`
+            Covariance matrix in the same order as the to be evaluated parameters.
+        """
+        tag = ["CovarianceGaussianPrior"]
+        def __init__(self, cov):
+            self.cov = cov
+            
+        def evaluate(self, values):
+            from numpy import linalg
+            return values @ linalg.inv(self.cov) @ values
+        
+        @property
+        def prior_parameters(self):
+            return [("cov", self.cov)]    
+            
+The prior classes are supposed to be set as ``.prior`` on:
+
+* Model-like classes, for instance the ``SkyModel`` class:
```

Comment:
The reason why it should be attached to the `Model` is bookkeeping. Priors should be serialized with the model. Parameters do not know, to which model they belong. The alternative is to handle it completely independent as we do for the covariance matrix and we would rely on the order of parameters. But in this case it would still be attached to the `Model`.